syntax = "proto3";

package petadoption;

service PetAdoptionService {
    rpc RegisterPet (Pet) returns (Response);
    rpc SearchPet (SearchRequest) returns (SearchResponse);
    rpc UpdatePet (Pet) returns (Response);  // Update a pet record
    rpc RemovePet (RemoveRequest) returns (Response);  // Remove a pet record
}

message Pet {
    string id = 1;  // Unique identifier for each pet
    string name = 2;
    string gender = 3;
    int32 age = 4;
    string breed = 5;
    bytes picture = 6;  // Storing the picture as bytes
}

message SearchRequest {
    string query = 1;  // Can be name, gender, age, or breed
}

message RemoveRequest {
    string id = 1;  // ID of the pet to remove
}

message Response {
    string message = 1;
}

message SearchResponse {
    repeated Pet pets = 1;  // List of matching pets
}